src/datafusion/scanner.rs

```diff
@@ -185,7 +185,7 @@ mod tests {
         let schema = first_batch.schema();
 
         let names = schema.fields().iter().map(|f| f.name()).collect::<Vec<_>>();
-        assert_eq!(names, vec!["lon", "lat"]);
+        assert_eq!(names, vec!["lat", "lon"]);
```

Comment:
I undid my change, restored the test to what it was before except that now it's checking for the last 2 columns in alphabetical order (so the columns are "float_data", "lat", "lon", and it's now picking up the last 2, always).